build-plugins/build-support/src/main/kotlin/aws/sdk/kotlin/gradle/plugins/artifactmetrics/AnalyzeMetricsTask.kt

```diff
@@ -0,0 +1,148 @@
+                ) { latestReleaseMetrics ->
+                    val remoteMetricsFile =
+                        project.layout.buildDirectory.file(OUTPUT_PATH + "latestReleaseMetrics.csv").get().asFile
+
+                    remoteMetricsFile.writeText(
+                        latestReleaseMetrics.body?.decodeToString() ?: throw GradleException("Metrics from latest release are empty"),
+                    )
+
+                    val remoteMetrics = remoteMetricsFile.toMap()
+                    val remoteLines = remoteMetricsFile.readLines()
+                    val remoteHeaders = remoteLines[0].split(",").map { it.trim() }
+
+                    val localMetrics = metricsFile.get().asFile.toMap()
+                    val localLines = metricsFile.get().asFile.readLines()
+                    val localHeaders = localLines[0].split(",").map { it.trim() }
+
+                    val artifactNames = remoteHeaders.toSet() + localHeaders.toSet()
+                    val metrics = mutableMapOf<String, DataPoint>()
+
+                    var addedArtifact = false
+                    var removedArtifact = false
+                    var significantChange = false
+
+                    artifactNames.forEach { artifact ->
+                        val local = localMetrics[artifact] ?: 0
+                        val remote = remoteMetrics[artifact] ?: 0
+
+                        val diff = local - remote
+                        val percentage = buildString {
+                            if (remote == 0L || local == 0L) {
+                                append("NaN")
+
+                                if (remote == 0L) addedArtifact = true
+                                if (local == 0L) removedArtifact = true
+
+                                return@buildString
+                            }
+
+                            if (diff < 0) append("-")
+
+                            val percentageValue = (abs(diff).toDouble() / remote.toDouble() * 100).roundToInt()
+                            if (percentageValue > 5 && diff > 0) significantChange = true
+
+                            append(percentageValue)
+                            append("%")
+                        }
+
+                        metrics[artifact] =
+                            DataPoint(
+                                remote,
+                                local,
+                                diff,
+                                percentage,
+                            )
+                    }
+
+                    shouldFailWorkflow.get().asFile.writeText(significantChange.toString())
+
+                    val diffTable = buildString {
+                        appendLine("Affected Artifacts\n=")
+                        appendLine("| Artifact |Pull Request (bytes) | Latest Release (bytes) | Delta (bytes / percentage) |")
+                        appendLine("| -------- | ------------------- | ---------------------- | -------------------------- |")
```

Comment:
**Nit**: Spacing out the header separators to match the header names is unnecessary since we're not padding the table body.

Reply:
Totally agree, I just did it for readability purposes. Should we get rid of it though?